### ISSUE TITLE
Verilog/VHDL API export

### ISSUE THREAD

**Issue by peppergrayxyz** [CircuitVerse/CircuitVerse#3336]
**Is your feature request related to a problem? Please describe.**
To export verilog the project needs to be opened from a browser, which is hard to automate.

**Describe the solution you'd like**
Hava an API to get the data, e.g. 
- GET /api/v1/projects/:id/verilog
- GET /api/v1/projects/:id/vhdl

**Describe alternatives you've considered**
There is no good workaround which would enable build scripts to fetch the data.

**Additional context**
https://tinytapeout.com/ has setup a toolchain with https://wokwi.com/ which works like this (verilog is exported by a github action and compiled). Repo: https://github.com/TinyTapeout/tt02-submission-template

**Are you working on this? (Yes/No)**
No

**Comment by vedant-jain03** [comment]
cc @gr455 @pulkit-30 @devartstar

**Comment by 092vk** [comment]
@tachyons Currently, the whole logic for generating the verilog code for a circuit is handled in the frontend by the browser. SoI  don't think we can implement this feature request until we mirror the verilog generation logic on the backend to expose it as an API.

**Comment by tachyons** [comment]
@092vk Yes, but this is something we can do in the future, That is why we kept this issue open

**Comment by Asrani-Aman** [comment]
can we add here GSoC 26 label

@092vk
